blockchain/blockchain.go

```diff
@@ -39,6 +54,17 @@ type Blockchain struct {
 	headersCache    *lru.Cache // LRU cache for the headers
 	difficultyCache *lru.Cache // LRU cache for the difficulty
 
+	// We need to keep track of block receipts between the verification phase
+	// and the insertion phase of a new block coming in. To avoid having to
+	// execute the transactions twice, we save the receipts from the initial execution
+	// in a cache, so we can grab it later when inserting the block.
+	// This is of course not an optimal solution - a better one would be to add
+	// the receipts to the proposed block (like we do with Transactions and Uncles), but
+	// that is currently not possible because it would break backwards compatibility due to
+	// insane conditionals in the RLP unmarshal methods for the Block structure, which prevent
```

Comment:
I would leave the caches for now, it would mess up the flow that's in place and just introduce backtracking for a PR that's already ready.

This PR needs to be viewed in the context it was created in, the fact that it wasn't reviewed in time is a different problem

Reply:
Agree, works for me